### PR TITLE
PIN-3576 Added eserviceId to PurposeUpdateContent

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**
It needs [PIN-3575](https://github.com/pagopa/interop-be-purpose-process/pull/169) to compile

[PIN-3575]: https://pagopa.atlassian.net/browse/PIN-3575?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala [hunk 1]
```diff
@@ -482,6 +482,8 @@ final case class PurposesApiServiceImpl(
 
     val result: Future[PurposeVersionResource] = for {
       purposeUUID    <- purposeId.toFutureUUID
+      _              <- catalogProcessService.getEServiceById(purposeUpdateContent.eserviceId)
+      _              <- tenantProcessService.getTenant(purposeUpdateContent.consumerId)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala]: [Same](https://github.com/pagopa/interop-be-purpose-process/pull/169/files#r1266751056)

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/PurposesApiServiceImpl.scala]: These are already checked at the process level.
We can remove them.

**Review by beetlecrunch — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -10494,6 +10494,12 @@ components:
           type: string
         description:
           type: string
+        eserviceId:
+          type: string
+          format: uuid
+        consumerId:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: as per other PRs, `consumerId` should not be changed

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -10503,6 +10506,8 @@ components:
       required:
         - title
         - description
+        - eserviceId
+        - consumerId
```
Comment (galales) [inline src/main/resources/interface-specification.yml:10510]:
```suggestion
```

**Review by galales — APPROVED** [review]

**Pull request merged**